src/vm/moar/dispatchers.nqp

```diff
@@ -3112,6 +3112,86 @@ nqp::dispatch('boot-syscall', 'dispatcher-register', 'raku-call-simple', -> $cap
     }
 });
 
+# Dispatcher to try to find a method, backing nqp::findmethod, nqp::tryfindmethod,
+# and nqp::can.
+nqp::dispatch('boot-syscall', 'dispatcher-register', 'raku-find-meth', -> $capture {
+    # See if this callsite is going megamorphic and do a fallback if so. We only do
+    # this in the non-exceptional case.
+    my $obj := nqp::captureposarg($capture, 0);
+    my $how := nqp::how_nd($obj);
+    my int $cache-size := nqp::dispatch('boot-syscall', 'dispatcher-inline-cache-size');
+    my int $exceptional := nqp::captureposarg_i($capture, 2);
+    if $cache-size >= 8 && !$exceptional && nqp::istype($how, Perl6::Metamodel::ClassHOW) {
```

Comment:
Because I measured some things, 16 seemed better, so I went for it instead of 8...and didn't update this location. :blush: Extracted these to named symbols instead now, as suggested. :-)